Firefox privacy notice for ToS (Fixes #15872)

alexgibson opened the pull request:
## One-line summary

Adds custom Firefox privacy notice and legal terms templates that will be displayed in-product in a modal view for the Firefox terms of service on-boarding experience.

## Significant changes and points to review

See requirements in the linked issue.

## Issue / Bugzilla link

#15872

## Testing

- http://localhost:8000/en-US/privacy/firefox/?v=product
- http://localhost:8000/en-US/about/legal/terms/firefox/?v=product

janbrasna commented:
I'm wondering if/how the in-product modal would support opening links away from the privacy notice? There is an internal link here and there to other privacy pages, but also a lot of external links in the expanded details sections to either sumo or wikimo, but also vendors and partners as onetrust, kevel, microsoft, google et al. — would these open from the modal in a new browser tab, or try to load in-place in the same view as the ToS? (i.e. it would have to support navigating back etc.)

alexgibson commented:
@janbrasna this is a good question, I'll try and find out the answer

janbrasna commented:
(To put in the context of a review: I was wondering whether this change had to also include some JS functionality to decorate any and all links to open in a blank target; something the common content does not do already by itself — in case the UI view would not allow/enable navigation, and/or handle links in its own expected way.)

The brief included "No need for a user to navigate anywhere else." — question if "no need" has some flexibility with pre-existing legal content including links within the texts… or the ask is to handle any further navigation in some way.

alexgibson commented:
@janbrasna Both our product and legal teams have discussed the internal link issue and are OK with the experience. There's no requirement to try and handle this ourselves on the website side at this time.

alexgibson commented on bedrock/privacy/views.py:
```diff
@@ -40,10 +40,13 @@ def get_legal_doc(self):
+        if variant == "product":
+            self.template_name = "privacy/notices/firefox-tos.html"
         else:
-            self.template_name = "privacy/notices/firefox-old-style-notice.html"
```
Note for reviewer: I removed this conditional as there does not seem to be any active locales in production that are using this older privacy notice template.

stephaniehobson reviewed (commented):
I haven't reviewed the CSS/formatting yet but I'll submit the suggestions I have so far.

Potential performance improvements:
- I noticed the page is still loading the consent banner HTML and JS. If I understand the requirements right that's not necessary. Likewise the `data` and `fxa` JS bundles.
- Since this is displaying in the Firefox theme you could remove the`m24-root` CSS bundle.
- Since there is no navigation the `m24-ui` JS bundle is not required.

alexgibson commented:
Updated the templates and naming convention to `firefox-simple` 👍 

I also added a commit to avoid loading unnecessary JS bundles. The firefox privacy notice required a few different things still for the page to work, so I made a new `firefox_privacy_simple` JS bundle containing only what's required (I think?).

stephaniehobson commented on media/css/privacy/privacy-firefox-simple.scss:
```diff
@@ -0,0 +1,186 @@
+    }
+}
+
+.format-paragraphs .privacy-body {
```
As far as I can tell none of the terms or notices use the `base-notice-paragraphs` template and `.format-paragraphs` styles can be safely omitted. (That template collapses content into [a series of "learn more" buttons](https://www.allizom.org/en-US/privacy/websites/)).

stephaniehobson reviewed (approved):
There's a little more code that can be deleted if you want but it's not a blocking improvement.

I think the "Choose how you want to share this data in Firefox." probably needs testing in-product but I assume this has to merge before that testing can happen.

R+ Thanks for the updates!
